src/client/src/services/pricingService.ts

```diff
@@ -1,61 +1,79 @@
 import { Observable, Scheduler } from 'rxjs/Rx'
-import { ServiceBase } from '../system/service'
+import { ServiceClient } from '../system/service'
+import { ServiceConst } from '../types'
 import { logger, RetryPolicy } from '../system'
 import '../system/observableExtensions/retryPolicyExt'
 
 const log = logger.create('PricingService')
 const getPriceUpdatesOperationName = 'getPriceUpdates'
 
-export default class PricingService extends ServiceBase {
-  cachedObservablesBySymbol = {}
+interface Request {
+  symbol: string,
+}
 
-  private createSpotPriceStream(request) {
-    return Observable.create(
-      (o) => {
-        log.debug(`Subscribing to spot price stream for [${request.symbol}]`)
-        let lastPrice = null
-        return this.serviceClient
-          .createStreamOperation(getPriceUpdatesOperationName, request)
-          // we retry the price stream forever, if it errors (likely connection down) we pump a non tradable price
-          .retryWithPolicy(
-            RetryPolicy.indefiniteEvery2Seconds,
-            getPriceUpdatesOperationName,
-            Scheduler.async,
-            (err, willRetry) => {
-              if (willRetry && lastPrice !== null) {
-                // TODO: remove is stale price fully supported
-              }
-            },
-          )
-          .map(item => keysToLower(item))
-          .subscribe(
-            (price) => {
-              lastPrice = price
-              o.next(price)
-            },
-            (err) => { o.error(err) },
-            () => { o.complete() },
-          )
+const createSpotPriceStream = (serviceClient, request: Request) => {
+  const observable = Observable.create(clientObserver => {
+    log.debug(`Subscribing to spot price stream for [${request.symbol}]`)
+    let lastPrice = null
+    const innerObserver = {
+      next: price => {
+        lastPrice = price
+        clientObserver.next(price)
       },
-    ).share()
-  }
+      error: err => {
+        clientObserver.error(err)
+      },
+      complete: () => {
+        clientObserver.complete()
+      }
+    }
+    const retryWithPolicyErrorCb = (err, willRetry) => {
+      if (willRetry && lastPrice !== null) {
+        // TODO: remove is stale price fully supported
+      }
+    }
+    const retryWithPolicyArgs = [
```

Comment:
Does this make it easier to read? It looks like indirection-by-alias to me.

Reply:
Yes, I think is easier to read the first code vs the second one. What do you think?

```js
    const subscription = serviceClient
      .createStreamOperation(getPriceUpdatesOperationName, request)
      // we retry the price stream forever, if it errors (likely connection down) we pump a non tradable price
      .retryWithPolicy(...retryWithPolicyArgs)
      .map(adaptDTO)
      .subscribe(innerObserver)
```

```js
    const subscription = serviceClient
      .createStreamOperation(getPriceUpdatesOperationName, request)
      // we retry the price stream forever, if it errors (likely connection down) we pump a non tradable price
      .retryWithPolicy(		
         RetryPolicy.indefiniteEvery2Seconds,		
         getPriceUpdatesOperationName,		
         Scheduler.async,		
         (err, willRetry) => {		
           if (willRetry && lastPrice !== null) {		
             // TODO: remove is stale price fully supported		
           }		
         },		
       )
      .map(adaptDTO)
      .subscribe(innerObserver)
```

Reply:
I have an easier time parsing a sequence of expressions than I do with the mental book-keeping that comes with assignments.

It was also the topic of the day yesterday, indirection that it is—the forms it takes and its trade offs.

I feel variable names are often too hopeful in what they will communicate—and the transparency which we hope they'll have. The names you've chosen are clear and concise, but for me the entailments are lost—I have to jump back and forth. It feels a lot like a computer paging when it runs out of memory 😅

Reply:
Depends at what level you read. I'm thinking that if you want to know what it does you should read like this:

```
   const subscription = serviceClient
      .createStreamOperation(getPriceUpdatesOperationName, request)
      .retryWithPolicy(...retryWithPolicyArgs)
      .map(adaptDTO)
      .subscribe(innerObserver)
```

If you are curious to know how certain thing is implemented, you click on it. I know it has some trade-offs, but normally you want to know what is doing from a higher point of view. I'm happy to change it if you find it too hard to read.

From the previous example you only lose this information `indefiniteEvery2Seconds`, but that could be added to the `retryWithPolicyArgs` variable.

Let me know your thoughts :)